platform-sdk/swirlds-tss/src/main/java/com/swirlds/tss/impl/groth21/FeldmanCommitment.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.tss.impl.groth21;
+
+import com.swirlds.pairings.api.FieldElement;
+import com.swirlds.pairings.api.Group;
+import com.swirlds.pairings.api.GroupElement;
+import com.swirlds.pairings.api.GroupElementAggregator;
+import com.swirlds.tss.api.TssShareId;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A Feldman polynomial commitment.
+ *
+ * @param commitmentCoefficients the commitment coefficients. Each element in this list consists of the group generator,
+ *                               raised to the power of a coefficient of the polynomial being committed to.
+ */
+public record FeldmanCommitment(@NonNull List<GroupElement> commitmentCoefficients) {
+    /**
+     * Creates a Feldman commitment.
+     *
+     * @param group      the group that elements of the commitment are in
+     * @param polynomial the polynomial to commit to
+     * @return the Feldman commitment
+     */
+    public static FeldmanCommitment create(@NonNull final Group group, @NonNull final DensePolynomial polynomial) {
+        final GroupElement generator = group.getGenerator();
+
+        final List<GroupElement> commitmentCoefficients = new ArrayList<>();
+        for (final FieldElement polynomialCoefficient : polynomial.coefficients()) {
+            commitmentCoefficients.add(generator.multiply(polynomialCoefficient));
+        }
+
+        return new FeldmanCommitment(commitmentCoefficients);
+    }
+
+    /**
+     * Extract the public key material for the given share ID.
+     * <p>
+     * The key material returned by this method will be aggregated with the public keys from all other commitments for
+     * the same share, and the result will be the final public key for that share.
+     *
+     * @param shareId the share ID of the public key to extract
+     * @return the public key extracted from this commitment
+     */
+    @NonNull
+    public GroupElement extractShareKeyMaterial(@NonNull final TssShareId shareId) {
+        if (commitmentCoefficients.size() < 2) {
+            throw new IllegalArgumentException("Coefficient commitments must have at least 2 elements");
+        }
+
+        //         for msg in dkg_messages.iter() {
+        //            // compute powers of share_id
```

Comment:
if we keep this commented code, could we have added comments explaining why and what it is?

Reply:
This commented code was just intended to make the rust code available for the review process. Not intended to stick around beyond that